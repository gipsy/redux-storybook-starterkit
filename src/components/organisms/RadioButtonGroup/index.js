import React from 'react'
import styled, { css } from 'styled-components'
import { font, palette } from 'styled-theme'
import PropTypes from 'prop-types'
import { rem, transitions } from 'polished'
import { responsiveBorderRadiuses as radius } from '~/src/components/themes/default'
import { responsiveBorderWidths as borderWidth } from '~/src/components/themes/default'
import { responsiveFontSizes as fontSize } from '~/src/components/themes/default'
import { responsiveButtonPaddings as padding } from '~/src/components/themes/default'
import { responsiveButtonHeights as height } from '~/src/components/themes/default'

import Field from '~/src/components/molecules/Field'

const styles = css`
  position: absolute;
  left: -99em;

  + span {
    font-family: ${font('primary')};
    font-weight: 600;
    width: 100%;
    display: block;
    cursor: default;
    margin-bottom: 20px;
    box-sizing: border-box;
    text-align: center;
    text-transform: uppercase;
    color: ${palette('grayscale', 1)};
    border-color: ${palette('grayscale', 2, true)};
    border-style: solid;
    ${transitions('border-color .1s ease-in')};

    @media (max-width: 420px) {
      font-size: ${fontSize.mobile};
      height: ${height.mobile};
      padding-top: ${padding.mobile};
      padding-bottom: ${padding.mobile};
      border-radius: ${radius.mobile};
      border-width: ${borderWidth.mobile};
    }

    @media (min-width: 421px) {
      font-size: ${fontSize.desktop};
      height: ${height.desktop};
      padding-top: ${padding.desktop};
      padding-bottom: ${padding.desktop};
      border-radius: ${radius.desktop};
      border-width: ${borderWidth.desktop};
    }

    &:hover {
      border-color: ${palette('grayscale', 3, true)};
    }
  }

  &:checked + span {
    background-color: ${palette('grayscale', 2)};
    color: ${palette('grayscale', 0, true)};
  }

  &:checked + span:hover {
    border-color: ${palette('grayscale', 2)};
  }
`

const RadioButtonElement = styled.input`${styles}`

const RadioButtonGroup = ({ name, listOfItems, ...props }) => {
  const renderItems = (item) => {
    return (
      <label key={item.value}>
        <RadioButtonElement type="radio" name='options' value={item.value} />
        <span>{item.text}</span>
      </label>
    )
  }
  return (
    <div>
      {listOfItems.map((item) => renderItems(item))}
    </div>
  )
}

RadioButtonGroup.propTypes = {
  listOfItems: PropTypes.array.isRequired,
  //selectedItemCallback: PropTypes.func.isRequired
}

export default RadioButtonGroup
