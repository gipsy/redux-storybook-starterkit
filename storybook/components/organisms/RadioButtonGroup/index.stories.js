import React from 'react'
import { storiesOf } from '@storybook/react'
import RadioButtonGroup from '~/src/components/organisms/RadioButtonGroup'

storiesOf('Components/Organisms', module)
  .addWithChapters(
    'RadioButtonGroup',
    {
      subtitle: 'Subtitle',
      info:`
        Radiobutton questions are found in our form as a type of input for our customers. We use the same format on both mobile and desktop
      `,
      chapters: [
        // List of chapters
        {
          title: 'RadioButtonGroup states',
          subtitle: 'subtitle',
          sections: [
            // List of sections
            {
              title: 'default',
              subtitle: 'subtitle',
              sectionFn: () => (
                <RadioButtonGroup listOfItems={
                    [
                      {
                        value: 'under 250',
                        text: 'under £250'
                      },
                      {
                        value: '250 - 500',
                        text: '£250 - £500'
                      },
                      {
                        value: '500 - 750',
                        text: '£500 - £750'
                      },
                      {
                        value: 'over 750',
                        text: 'over £750'
                      },
                    ]
                  }
                />
              ),
              options: {
                showSource: true,
                allowSourceToggling: true,
                showPropTables: false,
                allowPropTablesToggling: true,
              }
            },
            {
              title: 'checked',
              subtitle: 'checked state',
              sectionFn: () => (
                <RadioButtonGroup name='range_amount' listOfItems={
                    [
                      {
                        value: 'under 250',
                        text: 'under £250'
                      },
                      {
                        value: '250 - 500',
                        text: '£250 - £500'
                      },
                      {
                        value: '500 - 750',
                        text: '£500 - £750'
                      },
                      {
                        value: 'over 750',
                        text: 'over £750'
                      },
                    ]
                  }
                />
              ),
              options: {
                showSource: true,
                allowSourceToggling: true,
                showPropTables: false,
                allowPropTablesToggling: true,
              }
            },
          ]
        }
      ]
    }
  )
