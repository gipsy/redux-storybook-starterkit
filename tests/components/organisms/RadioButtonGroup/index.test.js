import React from 'react'
import { shallow } from 'enzyme'
import RadioButtonGroup from '~/src/components/organisms/RadioButtonGroup'

const wrap = (props = {}) => shallow(<RadioButtonGroup {...props} listOfItems={radioList.list} />)

const radioList = {
  list: [
    {value:'under 250',text:'under £250'},
    {value:'250 - 500',text:'£250 - £500'}
  ]
}

//#TODO test it but not on id as there are multiple radios
// it('renders props when passed in', () => {
//   const wrapper = wrap({ id: 'foo'})
//   expect(wrapper.find({ id: 'foo' })).toHaveLength(1)
// })

test('<RadioButtonGroup /> renders with list of buttons array', () => {
  const wrapper = wrap({
    id: 'test',
    palette: 'primary',
    listOfItems: radioList.list
  })
  expect(wrapper).toMatchSnapshot()
})
