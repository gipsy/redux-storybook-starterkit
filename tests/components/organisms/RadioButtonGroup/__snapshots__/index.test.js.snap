// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButtonGroup /> renders with list of buttons array 1`] = `
<div>
  <label>
    <styled.input
      name="options"
      type="radio"
      value="under 250"
    />
    <span>
      under £250
    </span>
  </label>
  <label>
    <styled.input
      name="options"
      type="radio"
      value="250 - 500"
    />
    <span>
      £250 - £500
    </span>
  </label>
</div>
`;
